examples/DotNetCore/WebDemoWithEventHub/WebDemoWithEventHub/EventHubService.cs

```diff
@@ -71,11 +63,7 @@ private void InitEventHubProcessor(EventHubConnection eventHubConnection)
 
         private Task ProcessEventHandler(ProcessEventArgs eventArgs)
         {
-            foreach (var refresher in _refreshers)
-            {
-                refresher.TryRefreshAsync();
-            }
-
+            _configRoot?.Reload();
```

Comment:
Just to add some context for reviewers, this `Reload` method is an existing method that will reconstruct the configuration from all configuration providers. @drago-draganov and I had a discussion and we feel that this would be a good starting point for a Reload method to include in our initial examples for consuming event grid events. We may plan to implement a similar method specific to our configuration provider in the future based on customer feedback.

Reply:
I could check but may have the answer faster :) ... Is `Reload` only available on `IConfigurationRoot` not `IConfiguration`?

Reply:
If events of 10 config changes all come in a very short period of time, are we going to reload the whole configuration (ours and others) 10 times? We throw the caching and the sentinel key (for configuration consistency) all out of the window?

Reply:
Yes, `Reload` method is available only in the derived class `IConfigurationRoot` but not in the base class `IConfiguration`.

---
In reply to: [432184784](https://github.com/Azure/AppConfiguration/pull/329#discussion_r432184784) [](ancestors = 432184784)

Reply:
Here's one way to look at this approach.

1. Pull and push refresh are de-coupled from each other. The `ConfigureRefresh` method is used to enable pull refresh and it allows the user to limit the settings to monitor as a part of refresh. Push refresh is based on events that pass the filter criteria for the event grid subscription. Receipt of such an event means that the user's intent is to update either one specific settings that corresponds to the event or all selected settings.

2. Users can configure their event grid subscription to raise events only for their sentinel key. A change in the sentinel key would result in Reload for their configuration in a consistent manner. So users can update any number of settings and then change the sentinel key to reload their configuration. What is the maximum number of *sentinel* changes we expect the users to make in the 99th percentile? Is it 10 or lower?

3. Whether we choose to de-couple pull and push refresh or not, users need to configure their event grid subscription with the appropriate filters. If we get 1000 events from event grid for different key-value changes, but the user has configured refresh only for a single key that was never updated, do we really want to allow users to invoke pull-based Refresh in this case?

---
In reply to: [432188646](https://github.com/Azure/AppConfiguration/pull/329#discussion_r432188646) [](ancestors = 432188646)

Reply:
There a number of hypotheses and "users need to"s here. I am not looking for a workaround. I am looking for a solution that is simple and intuitive.

This may not be the right place for this discussion. We should compare pros and cons if we go on a totally different route for push-updates.

Reply:
I agree with Abhilash that we could allow users to only listen for the sentinel key and refresh from that, but EventGrid/EventHub also helps us deal with issues of too many config changes in a short period of time. The event handler isn't invoked for each config change, but rather, the "ProcessEventArgs" argument contains a list of config changes since the last time event hub polled for changes. The exact polling interval isn't document, but based on my observation, it seems that there's usually a few seconds gap between each invocation of the event handler.

---
In reply to: [432612934](https://github.com/Azure/AppConfiguration/pull/329#discussion_r432612934) [](ancestors = 432612934,432188646)

Reply:
Used the Cache Invalidation approach we discussed.

---
In reply to: [432641835](https://github.com/Azure/AppConfiguration/pull/329#discussion_r432641835) [](ancestors = 432641835)